src/main/kotlin/org/agoranomic/assessor/lib/report.kt

```diff
@@ -119,11 +123,23 @@ fun StringBuilder.emitWithDelimiter(string: String) {
     emitLine("=".repeat(string.length))
 }
 
-data class ReportConfig(val voteComments: Boolean = true, val totalBallotCount: Boolean = true, val voteKindBallotCount: Boolean = true)
+data class ReportConfig(val json: Boolean = false, val voteComments: Boolean = true, val totalBallotCount: Boolean = true, val voteKindBallotCount: Boolean = true)
+
+val voteModule = SerializersModule {
+    polymorphic(Vote::class) {
+        InextricableVote::class with InextricableVote.serializer()
+        SimpleVote::class with SimpleVote.serializer()
+    }
+}
 
 fun report(resolutionMap: ProposalResolutionMap, config: ReportConfig = ReportConfig()): String {
     val sortedProposals = resolutionMap.proposals.sortedBy { it.number }
 
+    if (config.json){
```

Comment:
I think this should be an entirely separate function.

Reply:
Again, with you here—it's a little hacky. I couldn't quickly figure out a better architecture, but I'm sure there is one.